### PR TITLE
Add flop per joule

### PR DESCRIPTION
**Pull request by vatai**
Not sure where should `gflop_per_joule` be put. Now it is in the "root" of `result` i.e. `result["gflop_per_second"]`, but not sure if it should be in `result["power"]["gflop_per_joule"]`???

thinking

### CODE REVIEW

**Comment by vatai** [comment]
And I'm soooo happy I found out I can put images and gifs in the PRs!!!! Does it show??? 

And I also found this marvellous FF gif extension! https://addons.mozilla.org/en-US/firefox/addon/gifs-for-github/

**Comment by undertherain** [comment]
what's up?

File: benchmarker/__main__.py [hunk 1]
```diff
@@ -59,13 +59,15 @@ def main():
...
-            result["problem"]["gflop_measured"] = get_counters(command)
+            result["gflop_measured"] = get_counters(command)
         elif args.fapp_power:
             total, details = fapp.get_power_total_and_detail(command)
             result["problem"]["power"] = {"total": total, "details": details}
     cute_device = get_cute_device_str(result["device"]).replace(" ", "_")
     result["path_out"] = os.path.join(result["path_out"], result["problem"]["name"])
     result["path_out"] = os.path.join(result["path_out"], cute_device)
+    if "gflop_measured" in result.keys():
+        result["gflop_per_joule"] = result["gflop_measured"] / result["time_total"]
```
Comment (undertherain) [inline benchmarker/__main__.py:70]: that seems more like flops per second, not really per joule

**Review by undertherain — CHANGES_REQUESTED** [review]

File: benchmarker/__main__.py (hunk 1 above)
Comment (vatai) [reply benchmarker/__main__.py:70]: And that's the reason why we do reviews! (and the gifs of course). Any suggestion on where to put it?! Is it fine where it is?!

Comment (vatai) [reply benchmarker/__main__.py]: Done! Is that better? `"gflop_per_joule": 1.043047889268956,` with
```
python -m benchmarker --problem=resnet50 --framework=pytorch --problem_size=4 --batch_size=2 --nb_epoch=1 --flops
```

**Comment by undertherain** [comment]
copying from telegram
1. Most important is to get flop_per_sec

2. let's keep measured flop in int problem - to keep in consistent with estimated flop

3. let's check if astimated flop are available - well, here it is confusing anit, because if we have both estimated and measure flop - we have to decide which flop/sec to report  

4. you assume poewr measurement is always there - so it fails when power is not available 

```
[0]blackbird@sugi:~/Projects_heavy/performance/benchmarker$  [add-flop-per-joule|…13]$ python3 -m benchmarker --framework=pytorch --problem=resnet50 --problem_size=32 --batch_size=8 --nb_epoch=1 --precision=FP32 --flops
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/blackbird/Projects_heavy/performance/benchmarker/benchmarker/__main__.py", line 77, in <module>
    main()
  File "/home/blackbird/Projects_heavy/performance/benchmarker/benchmarker/__main__.py", line 71, in main
    result["gflop_per_joule"] /= float(result["power"]["joules_total"])
ZeroDivisionError: float division by zero
```

**Pull request merged**